Different cover integrations with the same error

### The problem

After updating to 2021.10.0b0 I have problems with any cover type I use. In my case OpenGarage and Template. One shows a problem in the logs and doesn't work. One just does nothing anymore. Config added too

### What is version of Home Assistant Core has the issue?

core-2021.10.0b0

### What was the last working version of Home Assistant Core?

core-2021.9.7

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

cover

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/cover/

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
Logger: homeassistant.bootstrap
Source: components/opengarage/cover.py:52
First occurred: 21:17:05 (1 occurrences)
Last logged: 21:17:05

Error setting up integration cover - received exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
    result = await async_setup_component(self.hass, domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
  File "/usr/src/homeassistant/homeassistant/setup.py", line 104, in async_setup_component
    return await task  # type: ignore
  File "/usr/src/homeassistant/homeassistant/setup.py", line 214, in _async_setup_component
    processed_config = await conf_util.async_process_component_config(
  File "/usr/src/homeassistant/homeassistant/config.py", line 887, in async_process_component_config
    platform = p_integration.get_platform(domain)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 532, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 537, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/usr/src/homeassistant/homeassistant/components/opengarage/cover.py", line 52, in <module>
    PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 739, in extend
    assert type(self.schema) == dict and type(schema) == dict, 'Both schemas must be dictionary-based'
AssertionError: Both schemas must be dictionary-based
```

```
cover:
  - platform: opengarage
    covers:
      garage:
        host: 192.168.181.43
        device_key: !secret garage_device_key
        name: Garage
  - platform: template
    covers:
      rolladengrosswohnzimmer:
        device_class: shade
        friendly_name: "Rollo Wohnzimmer"
        position_template: "{{ states('input_number.rollowohnzimmergr') > '0.0' }}"
        open_cover:
          service: switch.turn_on
          data:
            entity_id: switch.shellyuni_c45bbe5f8125_channel_1
        close_cover:
          service: switch.turn_on
          data:
            entity_id: switch.shellyuni_c45bbe5f8125_channel_2
        stop_cover:
          service: switch.turn_on
          data:
            entity_id: switch.shellyuni_c45bbe5f8125_channel_1
```

```
### Additional information

_No response_

I have similar error logs with my opengarage cover but components/tasmota/__init__.py is being thrown included?

```
2021-09-30 15:23:34 ERROR (MainThread) [homeassistant.bootstrap] Error setting up integration cover - received exception
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
result = await async_setup_component(self.hass, domain, self._hass_config)
File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
return await setup_tasks[domain] # type: ignore
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
result = await async_setup_component(self.hass, domain, self._hass_config)
File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
return await setup_tasks[domain] # type: ignore
File "/usr/src/homeassistant/homeassistant/components/tasmota/__init__.py", line 98, in start_platforms
await asyncio.gather(
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
result = await async_setup_component(self.hass, domain, self._hass_config)
File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
return await setup_tasks[domain] # type: ignore
File "/usr/src/homeassistant/homeassistant/setup.py", line 104, in async_setup_component
return await task # type: ignore
File "/usr/src/homeassistant/homeassistant/setup.py", line 214, in _async_setup_component
processed_config = await conf_util.async_process_component_config(
File "/usr/src/homeassistant/homeassistant/config.py", line 887, in async_process_component_config
platform = p_integration.get_platform(domain)
File "/usr/src/homeassistant/homeassistant/loader.py", line 532, in get_platform
cache[full_name] = self._import_platform(platform_name)
File "/usr/src/homeassistant/homeassistant/loader.py", line 537, in _import_platform
return importlib.import_module(f"{self.pkg_path}.{platform_name}")
File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
return _bootstrap._gcd_import(name[level:], package, level)
File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 850, in exec_module
File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
File "/usr/src/homeassistant/homeassistant/components/opengarage/cover.py", line 52, in <module>
PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 739, in extend
assert type(self.schema) == dict and type(schema) == dict, 'Both schemas must be dictionary-based'
AssertionError: Both schemas must be dictionary-based
2021-09-30 15:23:34 INFO (MainThread) [homeassistant.components.alarm_control_panel] Setting up alarm_control_panel.zha
2021-09-30 15:23:34 INFO (MainThread) [homeassistant.components.climate] Setting up climate.zha
2021-09-30 15:23:34 INFO (MainThread) [homeassistant.components.lock] Setting up lock.zha
2021-09-30 15:23:34 INFO (MainThread) [homeassistant.components.number] Setting up number.zha
2021-09-30 15:23:34 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
result = await async_setup_component(self.hass, domain, self._hass_config)
File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
return await setup_tasks[domain] # type: ignore
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
result = await async_setup_component(self.hass, domain, self._hass_config)
File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
return await setup_tasks[domain] # type: ignore
File "/usr/src/homeassistant/homeassistant/components/tasmota/__init__.py", line 98, in start_platforms
await asyncio.gather(
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1083, in async_forward_entry_setup
result = await async_setup_component(self.hass, domain, self._hass_config)
File "/usr/src/homeassistant/homeassistant/setup.py", line 97, in async_setup_component
return await setup_tasks[domain] # type: ignore
File "/usr/src/homeassistant/homeassistant/setup.py", line 104, in async_setup_component
return await task # type: ignore
File "/usr/src/homeassistant/homeassistant/setup.py", line 214, in _async_setup_component
processed_config = await conf_util.async_process_component_config(
File "/usr/src/homeassistant/homeassistant/config.py", line 887, in async_process_component_config
platform = p_integration.get_platform(domain)
File "/usr/src/homeassistant/homeassistant/loader.py", line 532, in get_platform
cache[full_name] = self._import_platform(platform_name)
File "/usr/src/homeassistant/homeassistant/loader.py", line 537, in _import_platform
return importlib.import_module(f"{self.pkg_path}.{platform_name}")
File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
return _bootstrap._gcd_import(name[level:], package, level)
File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 850, in exec_module
File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
File "/usr/src/homeassistant/homeassistant/components/opengarage/cover.py", line 52, in <module>
PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 739, in extend
assert type(self.schema) == dict and type(schema) == dict, 'Both schemas must be dictionary-based'
AssertionError: Both schemas must be dictionary-based
```

All tasmota devices are unavailable.

--- a/homeassistant/components/opengarage/config_flow.py
+++ b/homeassistant/components/opengarage/config_flow.py
@@ -60,13 +60,16 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     async def async_step_import(self, import_info):
         """Set the config entry up from yaml."""
-        import_info[CONF_HOST] = (
-            f"{'https' if import_info[CONF_SSL] else 'http'}://"
-            f"{import_info.get(CONF_HOST)}"
-        )
 
-        del import_info[CONF_SSL]
-        return await self.async_step_user(import_info)
+        user_input = {
+            CONF_DEVICE_KEY: import_info[CONF_DEVICE_KEY],
+            CONF_HOST: f"{'https' if import_info.get(CONF_SSL, False) else 'http'}://{import_info[CONF_HOST]}",
+            CONF_PORT: import_info.get(CONF_PORT, DEFAULT_PORT),
+            CONF_VERIFY_SSL: import_info.get(CONF_VERIFY_SSL, False),
+        }
+
+        import_info.pop(CONF_SSL, None)
+        return await self.async_step_user(user_input)
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None

--- a/homeassistant/components/opengarage/config_flow.py
+++ b/homeassistant/components/opengarage/config_flow.py
@@ -67,8 +67,6 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             CONF_PORT: import_info.get(CONF_PORT, DEFAULT_PORT),
             CONF_VERIFY_SSL: import_info.get(CONF_VERIFY_SSL, False),
         }
-
-        import_info.pop(CONF_SSL, None)
         return await self.async_step_user(user_input)
 
     async def async_step_user(

--- a/homeassistant/components/opengarage/cover.py
+++ b/homeassistant/components/opengarage/cover.py
@@ -58,7 +58,7 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     """Set up the OpenGarage covers."""
     _LOGGER.warning(
         "Open Garage YAML configuration is deprecated, "
-        "it has been imported into the UI automatically and can be safely removed."
+        "it has been imported into the UI automatically and can be safely removed"
     )
     devices = config.get(CONF_COVERS)
     for device_config in devices.values():

--- a/homeassistant/components/opengarage/cover.py
+++ b/homeassistant/components/opengarage/cover.py
@@ -49,17 +49,17 @@ COVER_SCHEMA = vol.Schema(
     }
 )
 
-PLATFORM_SCHEMA = cv.deprecated(
-    vol.All(
-        PLATFORM_SCHEMA.extend(
-            {vol.Required(CONF_COVERS): cv.schema_with_slug_keys(COVER_SCHEMA)},
-        )
-    )
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_COVERS): cv.schema_with_slug_keys(COVER_SCHEMA)}
 )
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the OpenGarage covers."""
+    _LOGGER.warning(
+        "Open Garage YAML configuration is deprecated, "
+        "it has been imported into the UI automatically and can be safely removed."
+    )
     devices = config.get(CONF_COVERS)
     for device_config in devices.values():
         hass.async_create_task(

--- a/homeassistant/components/opengarage/cover.py
+++ b/homeassistant/components/opengarage/cover.py
@@ -49,11 +49,12 @@ COVER_SCHEMA = vol.Schema(
     }
 )
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+PLATFORM_SCHEMA = cv.deprecated(
     vol.All(
-        cv.deprecated(DOMAIN),
-        {vol.Required(CONF_COVERS): cv.schema_with_slug_keys(COVER_SCHEMA)},
-    ),
+        PLATFORM_SCHEMA.extend(
+            {vol.Required(CONF_COVERS): cv.schema_with_slug_keys(COVER_SCHEMA)},
+        )
+    )
 )
 
 

